_posts/2017-5-31-release-1.3.0.md

```diff
@@ -0,0 +1,118 @@
+
+* **Fine-grained recovery** (beta): Instead of restarting the complete ExecutionGraph in case of a task failure, Flink is now able to restart only the affected subgraph and thereby significantly decrease recovery time ([FLINK-4256](https://issues.apache.org/jira/browse/FLINK-4256)).
+
+# DataStream API
+
+* **Side Outputs**: This change allows users to have more than one output stream for an operator. Operator metadata, internal system information (debugging, performance etc.) or rejected/late elements are potential use-cases for this new API feature. **The Window operator is now using this new feature for late window elements** ([FLINK-4460](https://issues.apache.org/jira/browse/FLINK-4460)).
+
+* **Union Operator State**: Flink 1.2.0 introduced broadcast state functionality, but this had not yet been exposed via a public API. Flink 1.3.0 provides the Union Operator State API for exposing broadcast operator state. The union state will send the entire state across all parallel instances to each instance on restore, giving each operator a full view of the state ([FLINK-5991](https://issues.apache.org/jira/browse/FLINK-5991)).
+
+* **Per-Window State**: Previously, the state that a WindowFunction or ProcessWindowFunction could access was scoped to the key of the window but not the window itself. With this new feature, users can keep window state independent of the key ([FLINK-5929](https://issues.apache.org/jira/browse/FLINK-5929)).
+
+# Deployment and Tooling
+
+* **Flink HistoryServer**: Flink’s [HistoryServer](https://ci.apache.org/projects/flink/flink-docs-release-1.3/monitoring/historyserver.html) now allows you to query the status and statistics of completed jobs that have been archived by a JobManager ([FLINK-1579](https://issues.apache.org/jira/browse/FLINK-1579)).
+
+* **Watermark Monitoring in Web Front-end**: For easier diagnosis of watermark issues, the Flink JobManager front-end now provides a new tab to track the watermark of each operator ([FLINK-3427](https://issues.apache.org/jira/browse/FLINK-3427)).
+
+* **Datadog HTTP Metrics Reporter**: Datadog is a widely-used metrics system, and Flink now offers a Datadog reporter that contacts the Datadog http endpoint directly ([FLINK-6013](https://issues.apache.org/jira/browse/FLINK-6013)).
+
+# Table API / SQL
+
+* **Support for Retractions in Table API / SQL**: Retraction is an important building block that will enable a whole range of new applications which require updating previously-emitted results. Examples for such use cases are computation of early results for long-running windows, updates due to late arriving data, or maintaining constantly changing results similar to materialized views in relational database systems. Flink 1.3.0 supports retraction for non-windowed aggregates. Results with updates can be either converted into a DataStream or materialized to external data stores using TableSinks with upsert or retraction support. 
```

Comment:
"As part of our endeavor to support continuous queries on Dynamic Tables, " and like to the blog post: http://flink.apache.org/news/2017/04/04/dynamic-tables.html